src/app/public/src/modules/sidebar/sidebar.component.ts

```diff
@@ -8,38 +8,39 @@ import { StacheRouteService } from '../shared';
   templateUrl: './sidebar.component.html',
   styleUrls: ['./sidebar.component.scss']
 })
-export class StacheSidebarComponent implements StacheNav, OnInit {
+export class StacheSidebarComponent implements StacheNav {
   @Input()
-  public routes: StacheNavLink[];
+  public set routes(value: StacheNavLink[]) {
+    this._routes = value ? value : this.routeService.getActiveRoutes();
+    this.childRoutes = this.filterRoutes(this._routes);
+  }
+
+  public get routes(): StacheNavLink[] {
+    return this._routes || [];
+  }
+
+  private _routes: StacheNavLink[];
   public heading: string;
   public headingRoute: string | string[];
   public childRoutes: StacheNavLink[];
 
-  public constructor(
-    private routeService: StacheRouteService) { }
-
-  public ngOnInit(): void {
-
-    if (!this.routes) {
-      this.routes = this.routeService.getActiveRoutes();
-    }
-
-    this.childRoutes = this.filterRoutes(this.routes);
-  }
+  public constructor(private routeService: StacheRouteService) { }
 
   public isHeadingActive(): boolean {
     const url = this.routeService.getActiveUrl();
     return (url === this.headingRoute);
   }
 
   private filterRoutes(routes: StacheNavLink[]): StacheNavLink[] {
-    const root = routes[0];
-    let headingPath = Array.isArray(root.path) ? root.path.join('/') : root.path;
-    headingPath = headingPath.replace(/^\//, '');
-    this.heading = root.name;
+    if (routes) {
```

Comment:
current tests don't hit this "else" case

Reply:
I think the routerService will always return a base route for the current path.  So this if check might not be necessary.